### ISSUE TITLE
Bad highlighting in synhtml-css-classes example

### ISSUE THREAD

**Issue by BenjaminRi** [trishume/syntect#317]
The command

```sh
cargo run --example synhtml-css-classes
```

produces the following HTML document:

![bad_highlighting](https://user-images.githubusercontent.com/28017860/100678288-1d3e3500-336d-11eb-9d8c-93ab4e9d85c8.png)

As we can see, the C++ output is incorrect. The color of `int` and also `}` is off. I tracked this down to:
```rust
let ss = SyntaxSet::load_defaults_newlines();
```

if I change it to

```rust
let ss = SyntaxSet::load_defaults_nonewlines();
```

it renders correctly:

![good_highlighting](https://user-images.githubusercontent.com/28017860/100678296-229b7f80-336d-11eb-8dad-8804a24acdd6.png)

However, before we charge in and do this simple little 2-character fix, we need to consider that the [documentation](https://docs.rs/syntect/4.4.0/syntect/parsing/struct.SyntaxSet.html#method.load_defaults_nonewlines) explicitly states that `load_defaults_nonewlines` is ["hacky" and "may not work" in some cases](https://docs.rs/syntect/4.4.0/syntect/parsing/struct.SyntaxSetBuilder.html#method.add_from_folder)

Therefore, I ask: Should we literally set a bad example and use the method that is not recommended? It is probably better to rewrite it such that it uses `load_defaults_newlines` yet works properly.

**Comment by trishume** [comment]
Oh yah sorry this is entirely because I haven't merged/released https://github.com/trishume/syntect/pull/307 for no good reason, mostly because I forgot about it.

**Comment by BenjaminRi** [comment]
> Oh yah sorry this is entirely because I haven't merged/released #307 for no good reason, mostly because I forgot about it.

Great, this patch and  `parse_html_for_line_which_includes_newline` exactly address my issues.

**Comment by BenjaminRi** [comment]
@trishume Would it be possible to release this on crates.io? Of course I can just depend on `master` on GitHub (or a fork of it), but that prevents me from publishing my own crate (which depends on `syntect`) on crates.io.

**Comment by trishume** [comment]
Sure sorry for the procrastination, released as `v4.5.0`.

**Comment by BenjaminRi** [comment]
I tested it and now it works (both the fixed example code and my own application using `v4.5.0`). I'll close my issues (including #318). Thank you for your work. No need to say sorry; you provided a valuable service for free.
